chainntnfs/interface.go

```diff
@@ -278,6 +278,34 @@ type SpendDetail struct {
 	SpendingHeight    int32
 }
 
+// HasSpenderWitness returns true if the spending transaction has non-empty
+// witness.
+func (s *SpendDetail) HasSpenderWitness() bool {
+	tx := s.SpendingTx
+
+	// If there are no inputs, then there is no witness.
+	if len(tx.TxIn) == 0 {
+		return false
```

Comment:
Nit: When we hit this case (I don't think ever) but should we not treat it as an error case as well and log it ?

Reply:
I think it's a legal state to have empty inputs here, guess it depends on the context of this method?